### ISSUE TITLE
off-by-one bug in omuse interface, forcing profiles

### ISSUE THREAD

**Issue by fjansson** [CloudResolvingClimateModeling/dales-omuse-examples#2]
Forcing profiles in Dales/fortran are 1-based while omuse index arrays are 0-based

all `set_tendency_*_` and `get_tendency_*_` should account for this.
```
 function set_tendency_U_(g_k, a, n) result(ret)
      integer, intent(in)                 :: n
      real, dimension(n), intent(in)      :: a
      integer, dimension(n), intent(in)   :: g_k
      integer                             :: ret
      !      u_tend(g_k(1:n)) = a(1:n)
             u_tend(g_k(1:n)+1) = a(1:n)
	                   
      ret = 0
    end function set_tendency_U_
```

**Comment by fjansson** [comment]
` get_profile_U_` also.
```
!  ret = gatherlayeravg(u0(2:i1, 2:j1, g_k(1:n)), a)
  ret = gatherlayeravg(u0(2:i1, 2:j1, g_k(1:n)+1), a)
```
`g_k` is zero-based, `u0` and other fields are indexed from 1 (when not counting ghost cells).

Optimization possibility: instead of indexing the 3D input field which may be expensive, pass the index array to the  `gatherlayeravg` function.

**Comment by ipelupessy** [comment]
fixed

**Comment by fjansson** [comment]
the fix we applied re-defined the index range of the grid in the omuse interface to start at 1 for the 1-d grids. I.e. we did not perform the change I suggested above.

The low-level get_profile_U which we use in sp-coupler were not affected, they used the correct 1-based indices.

Note that the index arrays for 3D fields start at 0, while index arrays for 1d fields start at 1, this is confusing. This can be changed later without breaking the interface.
